digitalocean/datasource_digitalocean_project_test.go

```diff
@@ -0,0 +1,64 @@
+package digitalocean
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceDigitalOceanProject_DefaultProject(t *testing.T) {
+	config := `
+data "digitalocean_project" "default" {
+}
+`
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.digitalocean_project.default", "id"),
+					resource.TestCheckResourceAttrSet("data.digitalocean_project.default", "name"),
+					resource.TestCheckResourceAttr("data.digitalocean_project.default", "is_default", "true"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceDigitalOceanProject_NonDefaultProject(t *testing.T) {
+	nonDefaultProjectName := randomName("tf-acc-project-", 6)
+	config := fmt.Sprintf(`
+resource "digitalocean_project" "foo" {
+	name = "%s"
+}
+
+data "digitalocean_project" "bar" {
+  	id = digitalocean_project.foo.id
+}
+
+data "digitalocean_project" "barfoo" {
+    name = "%s"
```

Comment:
This is failing for me. Think it's a race condition. It passes with this change to ensure the correct order of operations:

```suggestion
    name = digitalocean_project.foo.name
```

```
$ make testacc TESTARGS='-run=TestAccDataSourceDigitalOceanProject_NonDefaultProject'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccDataSourceDigitalOceanProject_NonDefaultProject -timeout 120m
?   	github.com/terraform-providers/terraform-provider-digitalocean	[no test files]
=== RUN   TestAccDataSourceDigitalOceanProject_NonDefaultProject
--- FAIL: TestAccDataSourceDigitalOceanProject_NonDefaultProject (0.18s)
    testing.go:640: Step 0 error: errors during refresh:
        
        Error: No projects found with name 'tf-acc-project-fv0132'
        
          on /tmp/tf-test053450873/main.tf line 10:
          (source code not available)
```